metadata-ingestion/src/datahub/ingestion/source/schema_inference/json.py

```diff
@@ -29,9 +29,24 @@
 
 
 class JsonInferrer(SchemaInferenceBase):
+    def parse_json(self, file):
+        complete_datastore = []
+        for line in file:
+            datastore = ujson.loads(line)
+            complete_datastore.append(datastore)
+
+        return complete_datastore
+
     def infer_schema(self, file: IO[bytes]) -> List[SchemaField]:
+        datastore = []
+
+        try:
+            file_dict = ujson.load(file)
+        except ValueError as ve:
+            file.seek(0)
+            file_dict = self.parse_json(file)
 
-        datastore = ujson.load(file)
+        datastore = file_dict
 
         if not isinstance(datastore, list):
```

Comment:
I suppose you can just embed it into a list on line 44 as we know that is where we will need it.